compiler/rustc_ast/src/visit.rs

```diff
@@ -925,6 +940,15 @@ pub fn walk_vis<'a, V: Visitor<'a>>(visitor: &mut V, vis: &'a Visibility) {
     }
 }
 
+pub fn walk_restriction<'a, V: Visitor<'a>, Kind: crate::RestrictionKind>(
+    visitor: &mut V,
+    restriction: &'a Restriction<Kind>,
+) {
+    if let RestrictionLevel::Restricted { ref path, id, shorthand: _ } = restriction.level {
```

Comment:
How likely is it that we'll gain new restrictions in the future? If it's likely, it might be better to do this as a match so the compiler helps us remember to update this.

Reply:
It's certainly a possibility. There was discussion of construct-only fields, which would not fit into the existing anywhere/in X/implied levels. So it's definitely something that can be changed. Both here and anywhere else in the PR.

Reply:
It is? Although I do see somewhere else in the PR that needs to be changed.